### PR TITLE
Refactor use of fields in job application flow

### PR DESCRIPTION
**Pull request by csutter**
- Move fields from `Wizardable` concern into form models
- Get rid of uses of `send` and `Wizardable`
- Use `ActiveModel::Naming` instead of manually building param keys

## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-3150

### CODE REVIEW

File: app/services/jobseekers/job_applications/quick_apply.rb [hunk 1]
```diff
@@ -28,7 +26,10 @@ def recent_job_application
   end
 
   def attributes_to_copy
-    personal_details_fields + professional_status_fields + employment_history_fields + ask_for_support_fields
+    ::Jobseekers::JobApplication::PersonalDetailsForm.fields +
```
Comment (david-mears-dfe) [inline app/services/jobseekers/job_applications/quick_apply.rb:29]: What does the initial `::` do?

Comment (csutter) [reply app/services/jobseekers/job_applications/quick_apply.rb]: I've removed them now, good shout. I briefly had this class definition wrapped into a module and that confused the constant lookup.

i.e. it was

Comment (csutter) [reply app/services/jobseekers/job_applications/quick_apply.rb]:
```ruby
module Jobseekers::JobApplications
  class QuickApply
    # ...

Comment (csutter) [reply app/services/jobseekers/job_applications/quick_apply.rb]: But basically `::` at the beginning says I explicitly intend this to be at the root of the namespace hierarchy, not nested within whatever module I'm in

File: app/form_models/jobseekers/job_application/personal_details_form.rb [hunk 2]
```diff
@@ -1,9 +1,13 @@
-class Jobseekers::JobApplication::PersonalDetailsForm
+class Jobseekers::JobApplication::PersonalDetailsForm < Jobseekers::JobApplication::BaseForm
   include ActiveModel::Model
 
   attr_accessor :city, :country, :email_address, :first_name, :last_name, :national_insurance_number,
```
[the remark this answers is not in the document]
Comment (csutter) [reply app/form_models/jobseekers/job_application/personal_details_form.rb:4]: Interesting!

File: app/controllers/jobseekers/job_applications/build_controller.rb [hunk 3]
```diff
@@ -37,20 +36,26 @@ def back_path
...
-      job_application.slice(*send("#{step}_fields"))
+      job_application.slice(form_class.fields)
     when "update"
       form_params
     end
   end
 
   def form_params
-    (params["jobseekers_job_application_#{step}_form".to_sym] || params).permit(*send("#{step}_fields"))
+    param_key = ActiveModel::Naming.param_key(form_class)
```
Comment (cesidio) [inline app/controllers/jobseekers/job_applications/build_controller.rb:56]: much nicer!

**Review by cesidio — APPROVED** [review]

**Pull request merged**